### ISSUE TITLE
1.3.5: Does not account for credit card number parts

### ISSUE THREAD

**Issue by aardrian** [w3c/wcag#2754]
Suggested outcomes:
1. Create a `cc-lastfour` autcomplete value; and/or
2. Explicitly state `cc-number` should not be used on fields that only request a part of a credit card number.

## Background
The current SC [1.3.5 Identify Input Purpose](https://www.w3.org/WAI/WCAG21/quickref/#identify-input-purpose) provides a [list of autocomplete values](https://www.w3.org/TR/WCAG21/#input-purposes) to satisfy the SC. One of them is `cc-number` number, but there is no value for only part of a credit card number.

## Context
It is relatively common for forms to ask for only the last 4 digits of a credit card number. Using the `cc-number` in some cases can present a potential barrier to some users successfully filling out the field.

## Details
- Consider a field with `maxlength="4"`.
- Neither the presence nor value of `maxlength` is conveyed to screen reader users.
- There is no autocomplete value for credit card number parts.
- Browsers will use the `cc-number` autocomplete value to fill the entire field.
- Firefox and Safari both allow all digits into the field.
- Chrome truncates the auto-filled value to the first four digits without immediately conveying this to screen reader users.
- Other users may also not notice this truncation (quick tests with distracted users show it was missed).

Test page: https://codepen.io/aardrian/pen/mdKyJWw

## Conclusions
Some audiences may struggle to clear the "last 4 digits" field, others may not know the wrong numbers were submitted, and some sub-sets of those users may struggle to get past this part of a form without clear error handling and messaging.

I don't believe it is a WCAG violation to _exclude_ `cc-number` on a field that only asks for part of the number. Further, using `cc-number` on a field like this can be an anti-pattern and should be avoided.

## Suggestion
1. Create a `cc-lastfour` autcomplete value; and/or
2. Explicitly state `cc-number` should not be used on fields that only request a part of a credit card number.

For the second suggestion, perhaps update [Technique H98: Using HTML 5.2 autocomplete attributes](https://www.w3.org/WAI/WCAG21/Techniques/html/H98) to revise [Example 1](https://www.w3.org/WAI/WCAG21/Techniques/html/H98#example-1-example-1-currency). Happy to do a PR, but want to be certain folks agree with this concern first.

**Comment by alastc** [comment]
Hi Adrian, 

In terms of our scope, this part:

> Create a cc-lastfour autcomplete value

Would be something for the HTML folks.

On the second part, that is an example we can add to the understanding, I don't see any controversy as a 4-number input it doesn't match the 'purpose' of the full CC number.

**Comment by aardrian** [comment]
I am happy to file an issue against HTML for the first part.

As for the second part, amending the Understanding doc, are you ok with me doing a PR?

**Comment by alastc** [comment]
Sure, a Pr would be good.

**Comment by patrickhlauke** [comment]
Going to close this for now, as we're currently bound by what's available as autocomplete values in HTML. When/if this is added, this can be revisited.
